src/notation/view/percussionpanel/percussionpanelpadlistmodel.cpp

```diff
@@ -79,6 +78,16 @@ void PercussionPanelPadListModel::deleteRow(int row)
     emit numPadsChanged();
 }
 
+void PercussionPanelPadListModel::removeEmptyRows()
+{
+    const int lastRowIndex = numPads() / NUM_COLUMNS - 1;
+    for (int i = lastRowIndex; i >= 0; --i) {
+        if (rowIsEmpty(i)) {
+            deleteRow(i);
```

Comment:
This can be optimized: we can find all empty rows and remove them all at once to prevent multiple layoutChanged() and numPadsChanged() calls